tests/integration/test_api/test_config/test_logs/test_api_logs_formats.py

```diff
@@ -0,0 +1,192 @@
+'''
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+type: integration
+brief: There is an API configuration option, called logs, which allows to log in 4 different ways ("json", "plain",
+       "json,plain", and "plain,json") through the format field. When the API is configured with one of those values the
+       logs are stored in the api.log and api.json files.
+tier: 2
+modules:
+    - api
+components:
+    - manager
+daemons:
+    - wazuh-apid
+os_platform:
+    - linux
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+references:
+    - https://documentation.wazuh.com/current/user-manual/api/configuration.html#logs
+tags:
+    - api
+    - logs
+    - logging
+'''
+import os
+import re
+
+import requests
+import pytest
+
+from wazuh_testing.tools import PREFIX, API_LOG_FILE_PATH, API_JSON_LOG_FILE_PATH
+from wazuh_testing.tools.configuration import get_api_conf
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.api import API_PROTOCOL, API_HOST, API_PORT, API_USER, API_PASS, API_LOGIN_ENDPOINT, \
+    API_GLOBAL_TIMEOUT, get_login_headers
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=2)]
+
+# Variables
+daemons_handler_configuration = {'all_daemons': True}
+test_directories = [os.path.join(PREFIX, 'test_logs')]
+error_log_payload = 'Timeout executing API request'
+
+# Configurations
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'configuration_api_logs_format.yaml')
+configurations = get_api_conf(configurations_path)
+tcase_ids = [f"level_{configuration['configuration']['logs']['level']}" +  \
+             f"_format_{configuration['configuration']['logs']['format']}" for  configuration in configurations]
+
+# Fixtures
+@pytest.fixture(scope='module', params=configurations, ids=tcase_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+
+    return request.param
+
+
+def send_request(login_attempts=5):
+    """Send a login request to the API."""
+
+    login_url = f"{API_PROTOCOL}://{API_HOST}:{API_PORT}{API_LOGIN_ENDPOINT}"
+
+    for _ in range(login_attempts):
+        response = requests.get(login_url, headers=get_login_headers(API_USER, API_PASS), verify=False,
+                                timeout=API_GLOBAL_TIMEOUT)
+        if response.status_code == 200: return True
```

Comment:
This should be an assertion.

Reply:
ok

Reply:
Done in ae04711e46bfb2b5dedbf08c5fafc046a9b4a61b, but later changed in bad335f469d52688db92b468b128a752b76a5568